test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetConsoleTestCase.cs

```diff
@@ -713,6 +712,84 @@ public async Task VerifyCacheFileInsideObjFolder(ProjectTemplate projectTemplate
             }
         }
 
+        [NuGetWpfTheory]
+        [MemberData(nameof(GetPackagesConfigTemplates))]
+        public async Task UpdateAllPCPackagesInPMC(ProjectTemplate projectTemplate)
+        {
+            EnsureVisualStudioHost();
+            using (var simpleTestPathContext = new SimpleTestPathContext())
+            {
+                // Arrange
+                var packageName1 = "UpdateAllPCPackage1";
+                var packageVersion1 = "1.0.0";
+                var packageVersion2 = "2.0.0";
+                await CommonUtility.CreatePackageInSourceAsync(simpleTestPathContext.PackageSource, packageName1, packageVersion1);
+                await CommonUtility.CreatePackageInSourceAsync(simpleTestPathContext.PackageSource, packageName1, packageVersion2);
+
+                var packageName2 = "UpdateAllPCPackage2";
+                var packageVersion3 = "1.0.1";
+                var packageVersion4 = "2.0.1";
+                await CommonUtility.CreatePackageInSourceAsync(simpleTestPathContext.PackageSource, packageName2, packageVersion3);
+                await CommonUtility.CreatePackageInSourceAsync(simpleTestPathContext.PackageSource, packageName2, packageVersion4);
+
+                using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, XunitLogger, addNetStandardFeeds: true, simpleTestPathContext: simpleTestPathContext))
+                {
+                    var solutionService = VisualStudio.Get<SolutionService>();
+                    var nugetConsole = GetConsole(testContext.Project);
+
+                    // Act
+                    nugetConsole.InstallPackageFromPMC(packageName1, packageVersion1);
+                    nugetConsole.InstallPackageFromPMC(packageName2, packageVersion3);
+                    nugetConsole.Execute("update-package");
+
+                    //Asset
+                    CommonUtility.AssertPackageInPackagesConfig(VisualStudio, testContext.Project, packageName1, packageVersion2, XunitLogger);
+                    CommonUtility.AssertPackageInPackagesConfig(VisualStudio, testContext.Project, packageName2, packageVersion4, XunitLogger);
+                    VisualStudio.AssertNuGetOutputDoesNotHaveErrors();
+                    Assert.True(VisualStudio.HasNoErrorsInOutputWindows());
+                }
+            }
+        }
+
+        [NuGetWpfTheory]
+        [MemberData(nameof(GetNetCoreTemplates))]
+        public async Task UpdateAllPRPackagesInPMC(ProjectTemplate projectTemplate)
```

Comment:
These 2 tests only different in the project template. 
They can be a single test, where we use MemberData as a means to provide the project templates.

Reply:
There is another difference: the assert methods. One is `AssertPackageInPackagesConfig` and the other is `AssertPackageReferenceExists`. That's why we wrote 2 tests for them.

Reply:
Ah, I see. 
Can we then use a common method for both? 
Potentially we can condition the assert based on the template.

Reply:
We have updated the case, please help to review it again. Thanks.